hpcpy/client/slurm.py

```diff
@@ -1,7 +1,216 @@
+        dry_run: bool = False,
+        depends_on: list = None,
+        delay: Union[datetime, timedelta] = None,
+        queue: str = None,
+        walltime: timedelta = None,
+        variables: dict = dict(),
+        **context,
+    ):
+        """Submit a job to the scheduler.
+
+        Parameters
+        ----------
+        job_script : Union[str, Path]
+            Path to the script.
+        directives : list, optional
+            List of complete directives to submit, by default list()
+        render : bool, optional
+            Render the job script from a template, by default False
+        dry_run : bool, optional
+            Return rather than executing the command, by default False
+        depends_on : list, optional
+            List of job IDs with successful exit on which this job depends, by default list()
+        delay: Union[datetime, timedelta]
+            Delay the start of this job until specific date or interval, by default None
+        queue: str, optional
+            Queue on which to submit the job, by default None
+        walltime: timedelta, optional
+            Walltime expressed as a timedelta, by default None
+        variables: dict, optional
+            Key/value environment variable pairs added to the qsub command.
+        **context:
+            Additional key/value pairs to be added to command/jobscript interpolation
+
+        Returns
+        -------
+        Job : hpcpy.job.Job
+            Job object.
+        """
+        # Get a logger from the parent
+        self._logger.debug(f"Submitting {job_script}")
+
+        # Initialise directives
+        directives = directives if isinstance(directives, list) else []
+
+        # Add job depends
+        if depends_on:
+
+            self._logger.debug("Job dependency specified.")
+
+            directives = self._interpolate_directive(
+                directives,
+                "depends_on",
+                depends_on_str=":".join(hu.ensure_list(depends_on)),
+            )
+
+        # Add delay (specified time or delta)
+        if delay:
+            self._logger.debug("Delay specified.")
+            delay_directive = self._assemble_delay_directive(delay)
+            directives.append(delay_directive)
+
+        # Add queue
+        if queue:
+            self._logger.debug(f"Queue specified ({queue})")
+            directives = self._interpolate_directive(directives, "queue", queue=queue)
+            context["queue"] = queue
+
+        # Add walltime
+        if walltime:
+            walltime_str = str(int(walltime.total_seconds() / 60.0))
+            self._logger.debug(f"Walltime specified ({walltime_str})")
+            directives = self._interpolate_directive(
+                directives, "walltime", walltime_str=walltime_str
+            )
+
+        # Update the environment with the variables (which is how Slurm does job variables)
+        self._logger.debug("Updating environment.")
+        _env = os.environ
```

Comment:
A copy rather than a reference will let you modify `_env` without altering the original `os.environ`, this is important for not leaking state to the end user
```suggestion
        _env = os.environ.copy()
```